sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/SqlQuerySpec.java

```diff
@@ -122,6 +123,26 @@ public SqlQuerySpec setParameters(List<SqlParameter> parameters) {
         return this;
     }
 
+    /**
+     * Adds the query parameter.
+     *
+     * @param parameter the query parameter.
+     * @return the SqlQuerySpec.
+     */
+    public SqlQuerySpec addSqlParameter(SqlParameter parameter) {
```

Comment:
we are adding public API. why can't this happen in the encryption layer rather than changing the core-sdk public surface area?

Reply:
Currently due to the lack of this api , it was very inefficient on encryption , to consume and recreate the array on every add encryption parameter. I don't see any reason for not having this api, please let me know if there is any which i am missing . Also it is a helpful api for regular cosmos user. Added beta annotation

Reply:
Removed the api